ansible_galaxy/models/collection_info.py

```diff
@@ -1,17 +1,79 @@
-
-import logging
+from __future__ import print_function
 
 import attr
+import json
+import logging
+import os
+import re
+import semver
 
-log = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
+TAG_REGEXP = re.compile('^[a-z0-9]+$')
 
 # see https://github.com/ansible/galaxy/issues/957
+
+
 @attr.s(frozen=True)
 class CollectionInfo(object):
-    namespace = attr.ib()
-    name = attr.ib()
-    version = attr.ib()
-    format_version = attr.ib(default=0.0)
-    author = attr.ib(default=None)
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+    authors = attr.ib(default=[])
     license = attr.ib(default=None)
+    description = attr.ib(default=None)
+    keywords = attr.ib(default=[])
+    readme = attr.ib(default='README.md')
+    dependencies = attr.ib(default=[])
+
+    @property
+    def namespace(self):
+        return self.name.split('.', 1)[0]
+
+    @name.validator
+    @version.validator
+    @license.validator
+    @description.validator
+    def _check_required(self, attribute, value):
+        if value is None:
+            raise ValueError("Invalid collection metadata. '%s' is required" % attribute.name)
+
+    @version.validator
+    def _check_version_format(self, attribute, value):
+        try:
+            semver.parse_version_info(value)
+        except ValueError:
+            raise ValueError("Invalid collection metadata. Expecting 'version' to be in "
```

Comment:
Does attr.ib end up showing the invalid value in this case? Poking at test_collection_info.test_semantic_version_error it seems like it doesn't. Ideally the error msg would indicate the context that is is the version field of the collection metadata as well as the bogus value. 

Could possibly a specific exception related to invalid versions with a 'value' attribute. ie `raise VersionValidationError(value=value)` so it could provide the invalid field value and a error message providing context about it being the version in the collection metadata.

Reply:
Included the value in the error message.